monkey/infection_monkey/monkey.py

```diff
@@ -158,13 +166,13 @@ def _connect_to_island_api(self) -> Tuple[Optional[str], Optional[IIslandAPIClie
             logger.info(f"Successfully connected to the island via {server}")
         else:
             raise Exception(
-                f"Failed to connect to the island via any known servers: {self._opts.servers}"
+                f"Failed to connect to the island via any known servers: {self._server_strings}"
             )
 
         # NOTE: Since we pass the address for each of our interfaces to the exploited
         # machines, is it possible for a machine to unintentionally unregister itself from the
         # relay if it is able to connect to the relay over multiple interfaces?
-        servers_to_close = (s for s in self._opts.servers if s != server and server_clients[s])
+        servers_to_close = (s for s in self._server_strings if s != server and server_clients[s])
```

Comment:
`server` is a SocketAddress, why do we need to compare it to string?

Reply:
Nice catch!